### PR TITLE
Remove stale pre-edit statuses from the thread view

### LINKED ISSUE #3366: Editing a reply in thread view duplicates that reply
Reported by a user via email, reproduces both in current nightly and Tusky 21.0.

Steps to reproduce:
- Open a post you replied to in thread view.
- Edit the reply
- The reply is now shown twice, once with the old unedited content, and once with the new content

Screenshot_20230221-200553

### PR DESCRIPTION
**Pull request by Tak**
Fixes #3366

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt [hunk 1]
```diff
@@ -85,6 +87,10 @@ abstract class SFragment : Fragment(), Injectable {
     @Inject
     lateinit var timelineCases: TimelineCases
 
+    private val composeLauncher = registerForActivityResult(ComposeResultContract()) { result ->
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt:90]: "result" is the status id here?

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt [hunk 2]
```diff
@@ -71,13 +71,13 @@ class ComposeViewModel @Inject constructor(
...
     private var startingContentWarning: String = ""
     private var inReplyToId: String? = null
-    private var originalStatusId: String? = null
     private var startingVisibility: Status.Visibility = Status.Visibility.UNKNOWN
 
     private var contentWarningStateChanged: Boolean = false
     private var modifiedInitialState: Boolean = false
     private var hasScheduledTimeChanged: Boolean = false
 
+    var originalStatusId: String? = null
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt:80]: This exposes the variable now?

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeResultContract.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.compose
+
+import android.content.Context
+import android.content.Intent
+import androidx.activity.result.contract.ActivityResultContract
+
+class ComposeResultContract : ActivityResultContract<ComposeActivity.ComposeOptions, String>() {
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeResultContract.kt:22]: Technical curiosity: Does this need a contract?

(The only other two usages I see are "take picture" and "login with browser" - so external stuff.)

Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeResultContract.kt:22]: It looked to me like it did? As in, android studio yelled at me that the old `startActivityForResult` was deprecated, and the docs led me to `ActivityResultContract` 🤷

File: app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt (hunk 1 above)
Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt:90]: yes

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt (hunk 2 above)
Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt:80]: Yes, so the activity can put it into the result

File: app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt (hunk 1 above)
Comment (Lakoja) [reply app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt:90]: Then how about "statusId"?

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt (hunk 2 above)
Comment (Lakoja) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt:80]: I see. That is done for other variables here as well...

**Comment by connyduck** [comment]
Wouldn't it be better to send a `StatusEditedEvent` similar to a `StatusComposedEvent`, instead of introducing a new mechanism? cc @nikclayton

**Comment by Tak** [comment]
Good point, I've reworked it to use the event hub mechanism

**Review by connyduck — APPROVED** [review]

**Pull request merged**